macros/answerHints.pl

```diff
@@ -146,11 +146,16 @@ sub AnswerHints {
       foreach my $wrong (@{$wrongList}) {
 	if (ref($wrong) eq 'CODE') {
 	  if (($ans->{score} < 1 || $options{checkCorrect}) &&
-	      ($ans->{ans_message} eq "" || $options{replaceMessage}) &&
-	      &$wrong($correct,$student,$ans)) {
-	    $ans->{ans_message} = $ans->{error_message} = $message;
-	    $ans->{score} = $options{score} if defined $options{score};
-	    last;
+	      ($ans->{ans_message} eq "" || $options{replaceMessage}) ) {
+	      # Make the call to run the function inside an eval to trap errors
+	      my $myResult = 0;
+	      eval { $myResult = &$wrong($correct,$student,$ans); };
+	      if ( $@ ) { $myResult = 0; }
```

Comment:
This line is not needed.  If the `$wrong` method has an error, then the assignment will not occur, and `$myResult` will still be `0` from the assignment on line 151.  Testing reveals that this is the case.

Reply:
Thanks. Dropped in commit 2 of the PR. However, I added a `or do` case which sets a fixed error message (in the results message box).